cluster-controller/src/main/java/io/strimzi/controller/cluster/resources/TopicController.java

```diff
@@ -0,0 +1,245 @@
+/*
+ * Copyright 2017-2018, Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.controller.cluster.resources;
+
+import io.fabric8.kubernetes.api.model.ConfigMap;
+import io.fabric8.kubernetes.api.model.EnvVar;
+import io.fabric8.kubernetes.api.model.EnvVarBuilder;
+import io.fabric8.kubernetes.api.model.extensions.Deployment;
+import io.strimzi.controller.cluster.ClusterController;
+import io.vertx.core.json.JsonObject;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/**
+ * Represents the topic controller deployment
+ */
+public class TopicController extends AbstractCluster {
+
+    public static final String KIND = "topic";
+
+    private static final String NAME_SUFFIX = "-topic-controller";
+
+    // Topic Controller configuration
+    protected TopicControllerConfig config;
+
+    // Port configuration
+    protected static final int HEALTHCHECK_PORT = 8080;
+    protected static final String HEALTHCHECK_PORT_NAME = "hcheck-port";
+
+    // Configuration defaults
+    protected static final String DEFAULT_IMAGE = "strimzi/topic-controller:latest";
+    protected static final int DEFAULT_REPLICAS = 1;
+    protected static final int DEFAULT_HEALTHCHECK_DELAY = 10;
+    protected static final int DEFAULT_HEALTHCHECK_TIMEOUT = 5;
+    protected static final int DEFAULT_ZOOKEEPER_PORT = 2181;
+    protected static final int DEFAULT_BOOTSTRAP_SERVERS_PORT = 9092;
+    protected static final String DEFAULT_FULL_RECONCILIATION_INTERVAL = "15 minutes";
+    protected static final String DEFAULT_ZOOKEEPER_SESSION_TIMEOUT = "20 seconds";
+
+    // Configuration keys
+    public static final String KEY_CONFIG = "topic-controller-config";
+
+    // Topic Controller configuration keys
+    public static final String KEY_KAFKA_BOOTSTRAP_SERVERS = "STRIMZI_KAFKA_BOOTSTRAP_SERVERS";
+    public static final String KEY_ZOOKEEPER_CONNECT = "STRIMZI_ZOOKEEPER_CONNECT";
+    public static final String KEY_NAMESPACE = "STRIMZI_NAMESPACE";
+    public static final String KEY_FULL_RECONCILIATION_INTERVAL = "STRIMZI_FULL_RECONCILIATION_INTERVAL";
+    public static final String KEY_ZOOKEEPER_SESSION_TIMEOUT = "STRIMZI_ZOOKEEPER_SESSION_TIMEOUT";
+
+    /**
+     * Constructor
+     *
+     * @param namespace Kubernetes/OpenShift namespace where cluster resources are going to be created
+     * @param cluster   overall cluster name
+     */
+    protected TopicController(String namespace, String cluster) {
+
+        super(namespace, cluster);
+        this.name = topicControllerName(cluster);
+        this.image = DEFAULT_IMAGE;
+        this.replicas = DEFAULT_REPLICAS;
+        this.healthCheckPath = "/";
```

Comment:
Really? We should use some other path to avoid problems should the tc grow an HTTP API.

Reply:
Actually the paths are `/healthy` and `/ready`. See the `generateDeployment` and `patchDeployment` which complete the paths. Before this class we consider having readiness and liveness on the same path for this reason only one `healthCheckPath` field. The current topic controller YAML has two different paths for them so we can chose to : using just one path as for the other cluster resources or implementing two different paths with two different variables for the probes.

Reply:
I think two fields and if they happen to be the same in some other cases that's not big deal.

Reply:
@tombentley I totally agree on that. Because this change involve all the other resources classes (starting from the `AbstractCluster` class) and not only the `TopicController` one, do we want to move this as an issue for a future PR ?

Reply:
An issue an future PR is fine with me